test/unit/controllers/registration/SubscriptionRecoveryControllerSpec.scala

```diff
@@ -144,6 +144,9 @@ class SubscriptionRecoveryControllerSpec
           any[SafeId]
         )(any[HeaderCarrier])
       ).thenReturn(Future.successful(result = ()))
+      when(mockSubscriptionDetailsHolder.dateOfBirth).thenReturn(None)
+      when(mockSubscriptionDetailsHolder.nameDobDetails)
+        .thenReturn(Some(NameDobMatchModel("fname", Some("mName"), "lname", LocalDate.parse("2019-01-01"))))
     }
```

Comment:
Do we need some tests here to assert that the right DoB is returned for all combinations of dateOfBirth / nameDobDetails?